packages/test/src/test-integration-update.ts

```diff
@@ -3,14 +3,31 @@ import { isGrpcServiceError } from '@temporalio/client';
 import * as wf from '@temporalio/workflow';
 import { helpers, makeTestFunction } from './helpers-integration';
 
-const test = makeTestFunction({ workflowsPath: __filename });
+// Use a reduced server long-poll expiration timeout, in order to confirm that client
+// polling/retry strategies result in the expected behavior
+const LONG_POLL_EXPIRATION_INTERVAL_SECONDS = 1.0;
+
+const test = makeTestFunction({
+  workflowsPath: __filename,
+  workflowEnvironmentOpts: {
+    server: {
+      extraArgs: [
+        '--dynamic-config-value',
+        `history.longPollExpirationInterval="${LONG_POLL_EXPIRATION_INTERVAL_SECONDS}s"`,
```

Comment:
You don't need to change this - changing the grpc timeout of the call itself will also make the server respond just before that with the empty reply.

That said, if this is easier, works too.

Reply:
> changing the grpc timeout of the call itself will also make the server respond just before that with the empty reply.

I haven't experimented to confirm the behavior, but that's not my understanding: the server responds with an empty response (indicating the most advanced stage known to have been reached) if it hits its own self-imposed deadline. But if the client's request timeout is reached then the server responds with the timeout error.

https://github.com/temporalio/temporal/blob/e7c9edd9732099770538a4ea2c4210847fcea6e1/service/history/api/updateworkflow/api.go#L272-L273